VmAgent.Core/Interfaces/FileSystemOperations.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Mono.Unix.Native;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace VmAgent.Core.Interfaces
+{
+    public class FileSystemOperations : IFileSystemOperations
+    {
+        public void AppendAllText(string path, string content)
+        {
+            File.AppendAllText(path, content);
+        }
+
+        public void CopyTo(FileInfo toCopy, string targetPath, bool overwrite)
+        {
+            toCopy.CopyTo(targetPath, overwrite);
+        }
+
+        public void Create(DirectoryInfo directoryInfo)
+        {
+            directoryInfo.Create();
+        }
+
+        public void CreateZipFile(string sourceDirectory, string destinationFilename)
+        {
+            ZipFile.CreateFromDirectory(sourceDirectory, destinationFilename);
+        }
+
+        public void Delete(FileSystemInfo fileSystemInfo)
+        {
+            DirectoryInfo dirInfo = fileSystemInfo as DirectoryInfo;
+            FileInfo fileInfo = fileSystemInfo as FileInfo;
+            if (dirInfo != null)
+            {
+                Directory.Delete(dirInfo.FullName, true);
+            }
+            else if(fileInfo != null)
+            {
+                File.Delete(fileInfo.FullName);
+            }
+            else
+            {
+                throw new ArgumentException("This is not a file or a directory");
+            }
+                
+        }
+
+        public bool Exists(FileSystemInfo fileSystemInfo)
+        {
+            return fileSystemInfo.Exists;
+        }
+
+        public void ExtractToDirectory(string sourcePath, string targetPath)
+        {
+            ZipFile.ExtractToDirectory(sourcePath, targetPath);
+        }
+
+        public IEnumerable<DirectoryInfo> GetDirectories(DirectoryInfo directoryInfo, bool recursive = false)
+        {
+            return directoryInfo.EnumerateDirectories("*", recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
+        }
+
+        public IEnumerable<FileInfo> GetFiles(DirectoryInfo directoryInfo, bool recursive = false)
+        {
+            return directoryInfo.EnumerateFiles("*", recursive ?  SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
+        }
+
+        public IEnumerable<FileSystemInfo> GetFileSystemInfos(DirectoryInfo directoryInfo, bool recursive = false)
```

Comment:
```suggestion
        public IEnumerable<FileSystemInfo> GetFileSystemInfo(DirectoryInfo directoryInfo, bool recursive = false)
```

Reply:
It's actually plural since it's returning the system info objects

Reply:
Its how the underlying call is named